### ISSUE TITLE
Mix of PO Box and Local Address for UPS

### ISSUE THREAD

**Issue by dbltoe** [lat9/multiship#37]
When any address is a PO Box, UPS is selected, and Continue is clicked; The cart returns to Step 1 with no error message shown to the customer and no error log is displayed in admin even with Enable debug? true.

If the customer realizes the error and reverts to USPS, they have to select USPS, click Continue (the cart will return to the same page), and click Continue a second time to get to Step 2.

This is the only operational glitch I can currently find.

**Comment by lat9** [comment]
I'll give that a look in a bit, but note that the plugin's _Debug_ option simply causes the multiship processing to occasionally burp out some 'trace' information for the debug of its operation.

**Comment by lat9** [comment]
Yes, that functionality got lost in the update.  I'm checking on that now.

**Comment by lat9** [comment]
That "should" do it (verified locally); let me know if it's still having issues!

**Comment by dbltoe** [comment]
Test site took all items (one each for three addresses) and listed them as UPS even with one being a PO Box.  No pause in operation or warning received.

**Comment by lat9** [comment]
What's the customer_id for that test customer?

**Comment by lat9** [comment]
My oh my!  That took some thought!  Essentially, the page-header processing for the `checkout_shipping` and `checkout_multiship` pages now have a couple of session-variables through which they communicate.

That allows the shipping processing to recognize that a change in the selected shipping method resulted in some previously-chosen ship-to addresses no longer being valid and also allows the multiship page to set an indication back to shipping so that, in this case, the customer has the opportunity to change their selected shipping method to one that 'supports' all their multiple addresses.

All changes have been uploaded to your test site, @dbltoe; I await your assessment and report.

**Comment by dbltoe** [comment]
Thank you so much!
Out on shopping spree with the wife. I’ll take a look later. 
John
